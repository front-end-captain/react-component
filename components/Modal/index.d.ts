import * as React from "react";
import * as Button from "./../Button/index.js";

export interface ModalProps {
  // 对话框是否可见
  visible?: boolean;

  // 确认按钮 loading
  confirmLoading?: boolean;

  // 标题
  title: string | React.ReactNode;

  // 是否显示右上角的关闭按钮
  closable: boolean;

  // 点击确认回调函数
  onOk?: (event: React.MouseEvent<any>) => void;

  // 点击取消的回调函数
  onCancel?: (event: React.MouseEvent<any>) => void;

  // 宽度
  width?: string | number;

  // 底部内容
  footer?: React.ReactNode;

  // 确认按钮文字
  okText?: string;

  // 取消按钮文字
  cancelText?: string;

  // 点击蒙层是否关闭模态框
  maskClosable?: boolean;

  // TODO 关闭后是否销毁模态框
  destroyOnClose?: boolean;

  prefixCls?: string;

  wrapClassName?: string;

  children?: React.ReactChildren;

  getPopupContainer?: () => void;

  zIndex?: number;

  // 蒙层样式
  maskStyle?: React.CSSProperties;

  // 是否显示蒙层
  mask?: boolean;

  // modal body 样式
  bodyStyle?: React.CSSProperties;

  // 可以设置浮层的样式、位置等
  style?: React.CSSProperties;

  // 是否垂直居中
  centered?: boolean;

  okButtonProps?: Button.ButtonProps;

  cancelButtonProps?: Button.ButtonProps;
}

export interface ModalState {
  firstMount: boolean;
  destroy: boolean;
}

export interface ModalMethodsProps {
  cancelText?: string;
  className?: string;
  content?: React.ReactNode;
  mask?: boolean;
  maskClosable?: boolean;
  okText?: string;
  prefixCls?: string;
  visible?: boolean;
  title?: React.ReactNode;
  // onOk?: () => void;
  onCancel?: () => void;
  width?: string | number;
  zIndex?: number;
  centered?: boolean;
  okButtonProps?: Button.ButtonProps;
  cancelButtonProps?: Button.ButtonProps;
}

export declare type ModalMethod = (props: ModalMethodsProps) => {
  destroy: () => void;
  update: (newConfig: ModalMethodsProps) => void;
};

declare class Modal extends React.Component<ModalProps, ModalState> {
  static info: ModalMethod;
  static success: ModalMethod;
  static error: ModalMethod;
  static warning: ModalMethod;
}

export default Modal;
