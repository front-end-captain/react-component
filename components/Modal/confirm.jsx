import React from "react";
import { render } from "react-dom";
import classNames from "classnames";

import Modal from "./modal.jsx";
import Button from "./../Button/index.js";

import { ConfirmModalWrapper } from "./modal.css.js";

const ConfirmModal = (props) => {
  const {
    prefixCls,
    className,
    content,
    mask,
    maskClosable,
    title,
    okText,
    visible,
    width,
    zIndex,
    type,
    icon,
    close,
    centered,
  } = props;

  const prefixClassName = prefixCls || "sky-modal";

  const contentPrefixCls = `${prefixClassName}-confirm`;

  const cls = classNames("sky-modal-confirm", `${contentPrefixCls}-${type}`, className);

  const config = {
    prefixCls: prefixClassName,
    contentPrefixCls,
    width: width || 416,
    mask: mask === undefined ? true : mask,
    maskClosable: maskClosable === undefined ? false : maskClosable,
    closable: false,
    title: "",
    footer: null,
    visible,
    wrapClassName: cls,
    zIndex,
    onCancel: close,
    centered,
  };

  return (
    <Modal {...config}>
      <ConfirmModalWrapper prefix={contentPrefixCls}>
        <div className={`${contentPrefixCls}-confirm-modal-body`}>
          <span className={`${contentPrefixCls}-title`}>
            {icon}
            {title}
          </span>
          <div className={`${contentPrefixCls}-content`}>{content}</div>
        </div>
        <div className={`${contentPrefixCls}-confirm-btn`}>
          <Button type="primary" onClick={close}>
            {okText || "知道了"}
          </Button>
        </div>
      </ConfirmModalWrapper>
    </Modal>
  );
};

const confirm = (config) => {
  const mountNode = document.createElement("div");
  document.body.appendChild(mountNode);

  let currentConfig = { ...config, visible: true };

  const renderConfirm = (props) => {
    render(<ConfirmModal {...props} />, mountNode);
  };

  const update = (newConfig) => {
    currentConfig = {
      ...currentConfig,
      ...newConfig,
    };

    renderConfirm(currentConfig);
  };

  const close = () => {
    currentConfig = {
      ...currentConfig,
      visible: false,
    };

    renderConfirm(currentConfig);
  };

  currentConfig = { ...currentConfig, close, visible: true };

  renderConfirm(currentConfig);

  return {
    destroy: close,
    update,
  };
};

export default confirm;
