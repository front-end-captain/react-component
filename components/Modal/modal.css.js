import styled from "styled-components";

import { skyMaskFadeIn, skyMaskFadeOut, skyFadeIn, skyFadeOut } from "./../_style/animate.js";
import { successColor, infoColor, warningColor, errorColor } from "./../_style/vars.js";

export const MaskWrapper = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.65);
  height: 100%;
  z-index: 1000;

  &.${(props) => props.prefixCls}-mask-hide {
    pointer-events: none;
    animation: ${skyMaskFadeOut} 0.5s linear forwards;
  }

  &.${(props) => props.prefixCls}-mask-show {
    pointer-events: auto;
    animation: ${skyMaskFadeIn} 0.5s linear forwards;
  }
`;

export const Wrapper = styled.div`
  position: fixed;
  overflow: auto;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1000;
  outline: 0;
  pointer-events: none;

  &.${(props) => props.prefixCls}-centered {
    text-align: center;

    &:before {
      content: "";
      display: inline-block;
      height: 100%;
      vertical-align: middle;
      width: 0;
    }

    .${(props) => props.prefixCls} {
      display: inline-block;
      top: 0;
      vertical-align: middle;
      text-align: left;
    }
  }

  &.sky-modal-confirm {
    .sky-modal-header,
    .sky-modal-footer {
      display: none;
    }
  }
`;

export const ModalWrapper = styled.div`
  pointer-events: auto;
  box-shadow: 0 4px 22px 0 rgba(15, 35, 95, 0.12);
  background-color: #fff;
  border-radius: 4px;
  width: ${(props) => props.width}px;
  max-height: calc(100vh - 48px);
  z-index: ${(props) => props.zIndex};
  transform-origin: bottom center;
  outline: none;
  position: relative;
  margin: 0 auto;
  padding: 0;
  list-style: none;
  transform: scale(0);
  top: 100px;

  &.${(props) => props.prefixCls}-open {
    pointer-events: auto;
    animation: ${skyFadeIn} 0.5s linear forwards;
  }

  &.${(props) => props.prefixCls}-close {
    pointer-events: none;
    animation: ${skyFadeOut} 0.5s linear forwards;
  }

  .sky-modal-content {
    .sky-modal-close {
      cursor: pointer;
      border: 0;
      background: transparent;
      position: absolute;
      right: 0;
      top: 0;
      z-index: 10;
      font-weight: 700;
      line-height: 1;
      text-decoration: none;
      transition: color 0.3s;
      color: rgba(0, 0, 0, 0.45);
      outline: 0;
      padding: 0;

      .sky-modal-close-container {
        display: block;
        font-style: normal;
        text-align: center;
        text-transform: none;
        text-rendering: auto;
        width: 56px;
        height: 56px;
        line-height: 56px;
        font-size: 20px;
      }
    }

    .sky-modal-header {
      padding: 16px 24px;
      border-radius: 4px 4px 0 0;
      background: #fff;
      color: rgba(0, 0, 0, 0.65);
      border-bottom: 1px solid #e8e8e8;

      .sky-modal-title {
        margin: 0;
        font-size: 16px;
        line-height: 22px;
        font-weight: 500;
        color: rgba(0, 0, 0, 0.85);
      }
    }

    .sky-modal-body {
      padding: 24px;
      font-size: 14px;
      line-height: 1.5;
      word-wrap: break-word;
    }

    .sky-modal-footer {
      border-top: 1px solid #e8e8e8;
      padding: 10px 16px;
      text-align: right;
      border-radius: 0 0 4px 4px;

      button + button {
        margin-left: 8px;
        margin-bottom: 0;
      }
    }
  }
`;

export const ConfirmModalWrapper = styled.div`
  .${(props) => props.prefix}-confirm-modal-body {
    .${(props) => props.prefix}-title {
      display: flex;
      color: rgba(0, 0, 0, 0.85);
      font-weight: 500;
      font-size: 16px;
      overflow: hidden;

      .sky-modal-confirm-icon {
        font-size: 22px;
        margin-right: 16px;
      }

      .sky-modal-confirm-icon-info {
        color: ${infoColor};
      }

      .sky-modal-confirm-icon-success {
        color: ${successColor};
      }

      .sky-modal-confirm-icon-error {
        color: ${errorColor};
      }

      .sky-modal-confirm-icon-warning {
        color: ${warningColor};
      }
    }

    .${(props) => props.prefix}-content {
      margin-left: 38px;
      font-size: 14px;
      color: rgba(0, 0, 0, 0.65);
      margin-top: 8px;
    }
  }

  .${(props) => props.prefix}-confirm-btn {
    margin-top: 24px;
    text-align: right;
  }
`;
