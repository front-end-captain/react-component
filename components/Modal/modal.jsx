import React from "react";
import { createPortal } from "react-dom";
import PropTypes from "prop-types";
import classNames from "classnames";

import confirm from "./confirm.jsx";
import Button from "./../Button/index.js";
import { CloseIcon, InfoIcon, SuccessIcon, ErrorIcon, WarningIcon } from "./../Icon/index.jsx";

import { MaskWrapper, Wrapper, ModalWrapper } from "./modal.css.js";

class Modal extends React.Component {
  static defaultProps = {
    prefixCls: "sky-modal",
    title: "",
    getPopupContainer: () => document.body,
    visible: false,
    confirmLoading: false,
    closable: true,
    width: 520,
    okText: "确认",
    cancelText: "取消",
    maskClosable: true,
    destroyOnClose: false,
    style: {},
    maskStyle: {},
    bodyStyle: {},
    zIndex: 999,
    mask: true,
    centered: false,
    okButtonProps: {},
    cancelButtonProps: {},
  };

  static propTypes = {
    prefixCls: PropTypes.string,
    visible: PropTypes.bool,
    wrapClassName: PropTypes.string,
    title: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
    getPopupContainer: PropTypes.func,
    confirmLoading: PropTypes.bool,
    closable: PropTypes.bool,
    width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    footer: PropTypes.oneOfType([PropTypes.instanceOf(null), PropTypes.element]),
    okText: PropTypes.string,
    cancelText: PropTypes.string,
    maskClosable: PropTypes.bool,
    /* eslint-disable-next-line */
    destroyOnClose: PropTypes.bool,
    style: PropTypes.objectOf(PropTypes.string),
    maskStyle: PropTypes.objectOf(PropTypes.string),
    bodyStyle: PropTypes.objectOf(PropTypes.string),
    zIndex: PropTypes.number,
    mask: PropTypes.bool,
    centered: PropTypes.bool,
    okButtonProps: PropTypes.objectOf(PropTypes.string),
    cancelButtonProps: PropTypes.objectOf(PropTypes.string),
  };

  static getDerivedStateFromProps({ visible, destroyOnClose }, { destroy, firstMount }) {
    if (visible) {
      return {
        firstMount: false,
        destroy: false,
      };
    }

    if (!visible && destroyOnClose && !destroy && !firstMount) {
      return {
        destroy: true,
      };
    }

    return null;
  }

  static iconClassName = "sky-modal-confirm-icon";

  static info = (props) => {
    const config = {
      type: "info",
      icon: <InfoIcon className={classNames(Modal.iconClassName, `${Modal.iconClassName}-info`)} />,
      ...props,
    };
    return confirm(config);
  };

  static success = (props) => {
    const config = {
      type: "success",
      icon: (
        <SuccessIcon
          className={classNames(Modal.iconClassName, `${Modal.iconClassName}-success`)}
        />
      ),
      ...props,
    };
    return confirm(config);
  };

  static error = (props) => {
    const config = {
      type: "error",
      icon: (
        <ErrorIcon className={classNames(Modal.iconClassName, `${Modal.iconClassName}-error`)} />
      ),
      ...props,
    };
    return confirm(config);
  };

  static warning = (props) => {
    const config = {
      type: "warning",
      icon: (
        <WarningIcon
          className={classNames(Modal.iconClassName, `${Modal.iconClassName}-warning`)}
        />
      ),
      ...props,
    };
    return confirm(config);
  };

  state = {
    firstMount: true,
    destroy: false,
  };

  onCancel = (event) => {
    const { onCancel } = this.props;
    if (onCancel) {
      onCancel(event);
    }
  };

  onOk = (event) => {
    const { onOk } = this.props;
    if (onOk) {
      onOk(event);
    }
  };

  renderFooter = () => {
    const {
      footer,
      confirmLoading,
      okText,
      cancelText,
      okButtonProps,
      cancelButtonProps,
    } = this.props;
    if (footer === null) {
      return null;
    }
    if (footer) {
      return footer;
    }

    return (
      <div className="sky-modal-footer-container">
        <Button onClick={this.onCancel} {...okButtonProps}>
          <span>{cancelText}</span>
        </Button>
        <Button type="primary" loading={confirmLoading} onClick={this.onOk} {...cancelButtonProps}>
          <span>{okText}</span>
        </Button>
      </div>
    );
  };

  render() {
    const { firstMount, destroy } = this.state;

    if (firstMount) {
      return null;
    }

    const {
      visible,
      prefixCls,
      wrapClassName,
      title,
      children,
      getPopupContainer,
      closable,
      width,
      maskClosable,
      style,
      maskStyle,
      bodyStyle,
      zIndex,
      mask,
      centered,
    } = this.props;

    const maskClassNames = classNames(`${prefixCls}-mask`, {
      [`${prefixCls}-mask-show`]: visible,
      [`${prefixCls}-mask-hide`]: !visible,
    });
    const wrapperClassNames = classNames(`${prefixCls}-wrapper`, wrapClassName, {
      [`${prefixCls}-centered`]: centered,
    });
    const ModalClassNames = classNames(`${prefixCls}`, {
      [`${prefixCls}-open`]: visible,
      [`${prefixCls}-close`]: !visible,
    });

    return createPortal(
      <div>
        {mask && (
          <MaskWrapper
            className={maskClassNames}
            prefixCls={prefixCls}
            onClick={maskClosable ? this.onCancel : undefined}
            style={{ ...maskStyle }}
          />
        )}

        <Wrapper className={wrapperClassNames} prefixCls={prefixCls}>
          {!destroy && (
            <ModalWrapper
              className={ModalClassNames}
              prefixCls={prefixCls}
              width={width}
              style={{ ...style }}
              zIndex={zIndex}
            >
              <div className="sky-modal-content">
                {closable && (
                  <button className="sky-modal-close" type="button" onClick={this.onCancel}>
                    <span className="sky-modal-close-container">
                      <CloseIcon />
                    </span>
                  </button>
                )}

                <div className="sky-modal-header">
                  <div className="sky-modal-title">{title}</div>
                </div>
                <div className="sky-modal-body" style={{ ...bodyStyle }}>
                  {children}
                </div>
                <div className="sky-modal-footer">{this.renderFooter()}</div>
              </div>
            </ModalWrapper>
          )}
        </Wrapper>
      </div>,
      getPopupContainer(),
    );
  }
}

export default Modal;
