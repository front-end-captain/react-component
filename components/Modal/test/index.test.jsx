import React from "react";
import { render } from "enzyme";
import assert from "power-assert";
import toJson from "enzyme-to-json";

import Modal from "./../index.js";

describe("<Modal />", () => {
  it("should render a Modal component", () => {
    const wrapper = render(
      <div>
        <Modal title="基本使用" visible>
          <p>some contents ...</p>
        </Modal>
        <Modal title="自定义取消按钮文字" cancelText="cancel" visible>
          <p>some contents ...</p>
        </Modal>
        <Modal title="没有 footer" visible footer={null}>
          <p>some contents ...</p>
        </Modal>
        <Modal title="水平垂直居中" visible centered>
          <p>some contents ...</p>
        </Modal>
      </div>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should return destroy method", () => {
    const modal = Modal.info();
    assert(modal.destroy && modal.destroy instanceof Function);
  });
});
