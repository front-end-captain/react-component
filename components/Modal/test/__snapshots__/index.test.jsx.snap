// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render a Modal component 1`] = `
<div>
  <div>
    <div
      class="sky-modal-mask sky-modal-mask-show modalcss__MaskWrapper-sc-1u1f9y2-0 lhqnkG"
    />
    <div
      class="sky-modal-wrapper modalcss__Wrapper-sc-1u1f9y2-1 gGxNTF"
    >
      <div
        class="sky-modal sky-modal-open modalcss__ModalWrapper-sc-1u1f9y2-2 eazxfY"
        width="520"
      >
        <div
          class="sky-modal-content"
        >
          <button
            class="sky-modal-close"
            type="button"
          >
            <span
              class="sky-modal-close-container"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
              >
                <line
                  x1="18"
                  x2="6"
                  y1="6"
                  y2="18"
                />
                <line
                  x1="6"
                  x2="18"
                  y1="6"
                  y2="18"
                />
              </svg>
            </span>
          </button>
          <div
            class="sky-modal-header"
          >
            <div
              class="sky-modal-title"
            >
              基本使用
            </div>
          </div>
          <div
            class="sky-modal-body"
          >
            <p>
              some contents ...
            </p>
          </div>
          <div
            class="sky-modal-footer"
          >
            <div
              class="sky-modal-footer-container"
            >
              <button
                class="sky-btn sky-btn-default sky-btn-normal Buttoncss__ButtonWrapper-k9s1tq-0 gbjZNk"
                type="button"
              >
                <span>
                  <span>
                    取消
                  </span>
                </span>
              </button>
              <button
                class="sky-btn sky-btn-primary Buttoncss__ButtonWrapper-k9s1tq-0 gbjZNk"
                type="button"
              >
                <span>
                  <span>
                    确认
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="sky-modal-mask sky-modal-mask-show modalcss__MaskWrapper-sc-1u1f9y2-0 lhqnkG"
    />
    <div
      class="sky-modal-wrapper modalcss__Wrapper-sc-1u1f9y2-1 gGxNTF"
    >
      <div
        class="sky-modal sky-modal-open modalcss__ModalWrapper-sc-1u1f9y2-2 eazxfY"
        width="520"
      >
        <div
          class="sky-modal-content"
        >
          <button
            class="sky-modal-close"
            type="button"
          >
            <span
              class="sky-modal-close-container"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
              >
                <line
                  x1="18"
                  x2="6"
                  y1="6"
                  y2="18"
                />
                <line
                  x1="6"
                  x2="18"
                  y1="6"
                  y2="18"
                />
              </svg>
            </span>
          </button>
          <div
            class="sky-modal-header"
          >
            <div
              class="sky-modal-title"
            >
              自定义取消按钮文字
            </div>
          </div>
          <div
            class="sky-modal-body"
          >
            <p>
              some contents ...
            </p>
          </div>
          <div
            class="sky-modal-footer"
          >
            <div
              class="sky-modal-footer-container"
            >
              <button
                class="sky-btn sky-btn-default sky-btn-normal Buttoncss__ButtonWrapper-k9s1tq-0 gbjZNk"
                type="button"
              >
                <span>
                  <span>
                    cancel
                  </span>
                </span>
              </button>
              <button
                class="sky-btn sky-btn-primary Buttoncss__ButtonWrapper-k9s1tq-0 gbjZNk"
                type="button"
              >
                <span>
                  <span>
                    确认
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="sky-modal-mask sky-modal-mask-show modalcss__MaskWrapper-sc-1u1f9y2-0 lhqnkG"
    />
    <div
      class="sky-modal-wrapper modalcss__Wrapper-sc-1u1f9y2-1 gGxNTF"
    >
      <div
        class="sky-modal sky-modal-open modalcss__ModalWrapper-sc-1u1f9y2-2 eazxfY"
        width="520"
      >
        <div
          class="sky-modal-content"
        >
          <button
            class="sky-modal-close"
            type="button"
          >
            <span
              class="sky-modal-close-container"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
              >
                <line
                  x1="18"
                  x2="6"
                  y1="6"
                  y2="18"
                />
                <line
                  x1="6"
                  x2="18"
                  y1="6"
                  y2="18"
                />
              </svg>
            </span>
          </button>
          <div
            class="sky-modal-header"
          >
            <div
              class="sky-modal-title"
            >
              没有 footer
            </div>
          </div>
          <div
            class="sky-modal-body"
          >
            <p>
              some contents ...
            </p>
          </div>
          <div
            class="sky-modal-footer"
          />
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="sky-modal-mask sky-modal-mask-show modalcss__MaskWrapper-sc-1u1f9y2-0 lhqnkG"
    />
    <div
      class="sky-modal-wrapper sky-modal-centered modalcss__Wrapper-sc-1u1f9y2-1 gGxNTF"
    >
      <div
        class="sky-modal sky-modal-open modalcss__ModalWrapper-sc-1u1f9y2-2 eazxfY"
        width="520"
      >
        <div
          class="sky-modal-content"
        >
          <button
            class="sky-modal-close"
            type="button"
          >
            <span
              class="sky-modal-close-container"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
              >
                <line
                  x1="18"
                  x2="6"
                  y1="6"
                  y2="18"
                />
                <line
                  x1="6"
                  x2="18"
                  y1="6"
                  y2="18"
                />
              </svg>
            </span>
          </button>
          <div
            class="sky-modal-header"
          >
            <div
              class="sky-modal-title"
            >
              水平垂直居中
            </div>
          </div>
          <div
            class="sky-modal-body"
          >
            <p>
              some contents ...
            </p>
          </div>
          <div
            class="sky-modal-footer"
          >
            <div
              class="sky-modal-footer-container"
            >
              <button
                class="sky-btn sky-btn-default sky-btn-normal Buttoncss__ButtonWrapper-k9s1tq-0 gbjZNk"
                type="button"
              >
                <span>
                  <span>
                    取消
                  </span>
                </span>
              </button>
              <button
                class="sky-btn sky-btn-primary Buttoncss__ButtonWrapper-k9s1tq-0 gbjZNk"
                type="button"
              >
                <span>
                  <span>
                    确认
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
